godot-core/src/obj/oneditor.rs

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use crate::meta::{FromGodot, GodotConvert, GodotType, PropertyHintInfo};
+use crate::registry::property::{DirectExport, Export, Var};
+
+/// Exported property that must be initialized in the editor (or associated code) before use.
+///
+/// Allows to use `Gd<T>`, which by itself never holds null objects, as an `#[export]` that should not be null during runtime.
+/// As such, it can be used as a more ergonomic way of `Option<Gd<T>>` which _assumes_ initialization.
+///
+/// Panics during access if uninitialized.
+/// When used inside a node class, `OnEditor` checks if a value has been set before `ready()` is run, and panics otherwise.
+/// Once initialized, it can be used almost as if it was a `T` value itself, due to `Deref`/`DerefMut` impls.
+///
+/// `OnEditor<T>` should always be used as a property, preferably in tandem with an `#[export]` or `#[var]`.
+/// Initializing `OnEditor` values via code before the first use is supported but should be limited to use cases involving builder or factory patterns.
+///
+/// [`Option<Gd<T>>`](std::option) and [`OnReady<Gd<T>>`](crate::obj::onready::OnReady) should be used for any other late initialization logic.
+///
+/// # Using `OnEditor<T>` with `Gd<T>` and `DynGd<T, D>`
+///
+/// ## Example - auto-generated init
+///
+/// ```
+///  use godot::prelude::*;
+///
+/// #[derive(GodotClass)]
+/// #[class(init, base = Node)]
+/// struct MyClass {
+///     #[export]
+///     editor_property: OnEditor<Gd<Resource>>,
+/// }
+///
+/// #[godot_api]
+/// impl INode for MyClass {
+///     fn ready(&mut self) {
+///         // Will always be valid and **must** be set via editor.
+///         // Additional check is being run before ready
+///         // to ensure that given value can't be null.
+///         let some_variant = self.editor_property.get_meta("SomeName");
```

Comment:
```suggestion
///         // Will always be valid and *must* be set via editor.
///         // Additional check is being run before ready() to ensure that given value can't be null.
///         let some_variant = self.editor_property.get_meta("SomeName");
```

Reply:
Sadly a manual wrap is required, outside of that I applied the suggestion fully :saluting_face: 
 
![image](https://github.com/user-attachments/assets/12821ded-4ea8-41db-9fb3-1b50228feda9)

Reply:
Thank you! :slightly_smiling_face: